backend/pixeldrain/pixeldrain.go

```diff
@@ -0,0 +1,605 @@
+// Package pixeldrain provides an interface to the Pixeldrain object storage
+// system.
+package pixeldrain
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/config/configmap"
+	"github.com/rclone/rclone/fs/config/configstruct"
+	"github.com/rclone/rclone/fs/fshttp"
+	"github.com/rclone/rclone/fs/hash"
+	"github.com/rclone/rclone/lib/rest"
+)
+
+const timeFormat = time.RFC3339Nano
+
+// Register with Fs
+func init() {
+	fs.Register(&fs.RegInfo{
+		Name:        "pixeldrain",
+		Description: "Pixeldrain Filesystem",
+		NewFs:       NewFs,
+		Config:      nil,
+		Options: []fs.Option{{
+			Name: "api_key",
+			Help: "API key for your pixeldrain account.\n" +
+				"Found on https://pixeldrain.com/user/api_keys.",
+			Sensitive: true,
+		}, {
+			Name: "directory_id",
```

Comment:
Note that this is called `root_folder_id` in many other backends. Can we call it that?

```
      --box-root-folder-id string          Fill in for rclone to use a non root folder as its starting point
      --drive-root-folder-id string        ID of the root folder
      --filefabric-root-folder-id string   ID of the root folder
      --onedrive-root-folder-id string     ID of the root folder
      --pcloud-root-folder-id string       Fill in for rclone to use a non root folder as its starting point (default "d0")
      --pikpak-root-folder-id string       ID of the root folder
      --sharefile-root-folder-id string    ID of the root folder
```